plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1813,16 +1947,25 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
       fc_elog(_log, "Not producing block because I don't have any private keys relevant to authority: ${authority}",
               ("authority", scheduled_producer.authority));
       _pending_block_mode = pending_block_mode::speculating;
+      not_producing_when_time = true;
    } else if (_pause_production) {
       fc_wlog(_log, "Not producing block because production is explicitly paused");
       _pending_block_mode = pending_block_mode::speculating;
+      not_producing_when_time = true;
    } else if (_max_irreversible_block_age_us.count() >= 0 && irreversible_block_age >= _max_irreversible_block_age_us) {
       fc_elog(_log, "Not producing block because the irreversible block is too old [age:${age}s, max:${max}s]",
               ("age", irreversible_block_age.count() / 1'000'000)("max", _max_irreversible_block_age_us.count() / 1'000'000));
       _pending_block_mode = pending_block_mode::speculating;
+      not_producing_when_time = true;
+   } else if (is_implicitly_paused()) {
+      fc_elog(_log, "Not producing block because no recent votes, last producer vote ${pv}, other votes ${ov}, last block time ${bt}",
+              ("pv", _last_producer_vote_received.load(std::memory_order_relaxed))
+              ("ov", _last_other_vote_received.load(std::memory_order_relaxed))("bt", _accepted_block_time));
+      _pending_block_mode = pending_block_mode::speculating;
+      not_producing_when_time = true;
    }
 
-   if (in_speculating_mode()) {
+   if (in_speculating_mode() && !not_producing_when_time) {
```

Comment:
why ` && !not_producing_when_time`?

Reply:
`!not_producing_when_time to avoid tight spin because of error or paused production` I'll add a comment